pkg/plugins/backendplugin/grpcplugin/client_proto.go

```diff
@@ -0,0 +1,126 @@
+package grpcplugin
+
+import (
+	"context"
+	"errors"
+
+	goplugin "github.com/hashicorp/go-plugin"
+	"google.golang.org/grpc"
+
+	"github.com/grafana/grafana-plugin-sdk-go/genproto/pluginv2"
+
+	"github.com/grafana/grafana/pkg/plugins/log"
+)
+
+var (
+	errClientNotStarted = errors.New("plugin client has not been started")
+)
+
+var _ ProtoClient = (*protoClient)(nil)
+
+type ProtoClient interface {
+	pluginv2.DataClient
+	pluginv2.ResourceClient
+	pluginv2.DiagnosticsClient
+	pluginv2.StreamClient
+
+	PluginID() string
+	Logger() log.Logger
+	Start(context.Context) error
+	Stop(context.Context) error
+}
+
+type protoClient struct {
+	plugin *grpcPlugin
+}
+
+type ProtoClientOpts struct {
+	PluginID       string
+	ExecutablePath string
+	ExecutableArgs []string
+	Env            []string
+	Logger         log.Logger
+}
+
+func NewProtoClient(opts ProtoClientOpts) (ProtoClient, error) {
+	p := &grpcPlugin{
```

Comment:
Could you use `newPlugin` here to reuse code?

Reply:
Do you mean the `newPlugin` func that returns a `backendplugin.Plugin` interface?

Reply:
Yes that was my idea. But we discussed offline and feel free to merge to move this forward